tests/glsl-intrinsic/atomic/atomicCounter.slang

```diff
@@ -0,0 +1,142 @@
+//TEST:SIMPLE(filecheck=CHECK_GLSL):  -allow-glsl -stage compute -entry computeMain -target glsl -DTARGET_GLSL
+//TEST:SIMPLE(filecheck=CHECK_SPV):  -allow-glsl -stage compute -entry computeMain -target spirv -emit-spirv-directly -DTARGET_SPIRV
+
+// currently tests do not handle atomic_uint->storage-buffer too well (mapping to binding location)
+//DISABLE_TEST:TEST(compute, vulkan):COMPARE_COMPUTE(filecheck-buffer=BUF):-vk -compute -entry computeMain -allow-glsl
+//DISABLE_TEST:TEST(compute, vulkan):COMPARE_COMPUTE(filecheck-buffer=BUF):-vk -compute -entry computeMain -allow-glsl -emit-spirv-directly
+#version 430
+
+//TEST_INPUT:ubuffer(data=[0 0 0 0 0 0], stride=4):out
+layout(binding = 1, offset = 12) uniform atomic_uint one;
+layout(binding = 1) uniform atomic_uint two;
+
+//TEST_INPUT:ubuffer(data=[0 0], stride=4):out,name=outputBuffer
+buffer MyBlockName
+{
+    uint data[2];
+} outputBuffer;
+
+bool testSetterAndGetter()
+{
+    return true
+// CHECK_GLSL-DAG-N: atomicCounterExchange
+// CHECK_SPV-DAG-N: OpAtomicExchange
```

Comment:
DAG-N means, if the following token shows up in any order at least once+previous counts of the DAG-N number of times then your token is found

needed since atomicCounterExchange will show up more than once and the function will likely be mangled some amount.

 LLVM FileCheck has a lot of undocumented/barley-documented components

Reply:
I will be careful to use rare syntax of FileCheck.
You also need to double check those syntax actually work with the version we use.
If I do something like
```
//CHECK_GLSL-DAG-MMM:
```
It will simply be considered as a comment and it wouldn't do anything.

Reply:
I noticed the following and removed all of the not properly working expressions